src/lib/style-properties.html

```diff
@@ -331,6 +334,9 @@
           // parsedSelector fallback for 'shady' css build
           selectorToMatch = rule.transformedSelector || rule.parsedSelector;
         }
+        if (isRoot && hostScope === 'html') {
```

Comment:
I think what you have makes sense. I missed the fact that we start with matching the host scope and then only conditionally match the others

And... won't `:root` in an element now be transformed to `:host > *`?